### DISCUSSION TITLE
Non-linear unit conversion (Watts, dBW, dBm)

### DISCUSSION THREAD

**Discussion by jacob031** [josdejong/mathjs#2584] [Q&A]
Basically, I need to implement support for _Watts_ to _dBW_ and _dBm_. The conversion from _Watts_ to _dBW_ is defined as _P(dBW) = 10*log10( P(w) / 1W )_. Implementing this conversion using the options provided with the _mathjs.createUnit_ function just doesn't seem possible.

I've had no luck in finding an alternative solution by searching various forums, the discussions here on Github, and the documentation. I feel like I am missing something obvious.

Any advice?

**Comment by josdejong** [comment]
Ah, yeah, I think it's not possible right now to define non-linear unit conversions. 

As a workaround you could maybe implement a helper function like `wattToDBW(...)`.

@ericman314 do you have any thoughts in this regard?
